lib/SftpClient.js

```diff
@@ -0,0 +1,49 @@
+const SFTP = require('sftp-promises')
+const FileParser = require('ftp/lib/parser')
+
+class SftpClient {
+  constructor ({ host, port = 21, user, password }) {
+    this.host = host
+    this.port = port
+    this.user = user
+    this.password = password
+    this.client = new SFTP()
+  }
+
+  async connect () {
+    this.session = await this.client.session({
+      host: this.host,
+      port: this.port,
+      username: this.user,
+      password: this.password
+    })
+
+    return this.session
+  }
+
+  async disconnect () {
+    if (this.session !== null) {
```

Comment:
It should be replaced by `if (this.session) {`. That's a more common pattern in JS to also catch `undefined`, uninitialized variables.